### PR TITLE
Reconnect

### PR DESCRIPTION
**Pull request by lesniakania**

### CODE REVIEW

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt [hunk 1]
```diff
@@ -240,6 +272,21 @@ abstract class SessionViewMvcImpl<ListenerType>: BaseObservableViewMvc<ListenerT
         mLoader?.visibility = View.GONE
     }
 
+    override fun showReconnectingLoader() {
+        AnimatedLoader(mReconnectingLoader).start()
+        mReconnectingLoader.visibility = View.VISIBLE
+        mReconnectButton.isEnabled = false
+        mReconnectButton.translationY = 6f
+        mReconnectButton.translationZ = -6f
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt:280]: I know.... 😂

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/charts/ChartAveragesCreator.kt [hunk 2]
```diff
@@ -26,6 +26,8 @@ class ChartAveragesCreator {
         val measurements: MutableList<Measurement>? =
             stream.getMeasurementsForPeriod(MAX_AVERAGES_AMOUNT, MOBILE_FREQUENCY_DIVISOR)
 
+        if (measurementsInPeriod == 0) return mutableListOf()
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/charts/ChartAveragesCreator.kt:29]: this was a kind of weird bug that I experienced randomly that measurementsInPeriod was 0...

File: app/src/main/java/io/lunarlogic/aircasting/screens/new_session/select_device/DeviceItem.kt [hunk 3]
```diff
@@ -16,7 +16,7 @@ open class DeviceItem(private val mBluetoothDevice: BluetoothDevice) {
     init {
         name = mBluetoothDevice.name ?: "Unknown"
         address = mBluetoothDevice.address
-        id = name.split(":").last()
+        id = name.split(":", "-").last()
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/screens/new_session/select_device/DeviceItem.kt:19]: this is part of the fix for AB1

File: app/src/main/java/io/lunarlogic/aircasting/sensor/AirBeamConnectorFactory.kt [hunk 4]
```diff
@@ -16,7 +16,7 @@ open class AirBeamConnectorFactory(
         when(deviceItem.type) {
             DeviceItem.Type.AIRBEAM2 -> return AirBeam2Connector(mSettings, mErrorHandler)
             DeviceItem.Type.AIRBEAM3 -> return AirBeam3Connector(mContext, mSettings, mErrorHandler)
-            else -> return null
+            else -> return AirBeam2Connector(mSettings, mErrorHandler)
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/sensor/AirBeamConnectorFactory.kt:19]: this is second part of the fix for AB1

File: app/src/main/java/io/lunarlogic/aircasting/database/converters/SessionStatusListConverter.kt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+            }
+
+            return statuses
+        }
+
+        fun toIntList(statuses: List<Session.Status>): List<Int> {
+            return statuses.map { it.value }
+        }
+    }
+}
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/database/converters/SessionStatusListConverter.kt:21]: I actually ended up not using this, so I removed it along with some unused imports here: https://github.com/HabitatMap/AircastingAndroid/commit/f1df9216af5d91818f8cb97d97bae2a29c6099ae

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt (hunk 1 above)
Comment (mturnau) [reply app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt:280]: comment / separate style / constants

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/charts/ChartAveragesCreator.kt (hunk 2 above)
Comment (mturnau) [reply app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/charts/ChartAveragesCreator.kt:29]: charts on disconnect session… - after 1 min of recording, I reconnected, waited 1 min and once - I got all 9 mins of chart (even though there was no 9 minutes of recording), second time - I had one dot all the time

**Pull request merged**